lib/bots/versionbot.ts

```diff
@@ -802,22 +808,44 @@ export class VersionBot extends GithubBot.GithubBot {
                 repo
             });
         }).then((statuses: GithubBotApiTypes.CombinedStatus) => {
-            // We go through the statuses and check them off against the contexts.
-            // If we get to the end and:
-            //  1. Not all of the contexts have been seen
-            //  2. Any of them have a state other than 'success'
-            // Then the status checks have failed and we throw an error.
-            _.each(statuses.statuses, (status) => {
-                if (_.includes(contexts, status.context)) {
-                    if (status.state === 'success') {
-                        foundStatuses.push(true);
-                    } else {
-                        foundStatuses.push(false);
+            // Contexts need to be checked specifically.
+            // Branch protection can include contexts that use prefixes which are then
+            // suffixed to create more statuses.
+            // For example, 'continuous-integration/travis-ci' contexts can end up as:
+            //  * continuous-integration/travis-ci/push
+            //  * continuous-integration/travis-ci/pr
+            // statuses, which mean there are actually two checks per context and not one.
+            //
+            // The simplest way to check the contexts are therefore to get a list of
+            // required status contexts (which we do anyway), then go through each
+            // actual status check from the combined, and try and match the prefix of a context
+            // with each status. If we get a hit, and the status is a failure, then we
+            // have failed. If we match and the status is a pass, we've passed.
+            // We can therefore assume that a pass has occurred if:
+            //  * We have seen one of every context in the protected status list at least once
+            //    AND
+            //  * Each of those seen has passed
+            // Should any protected context not be seen in the current status checks, then
+            // we have failed.
```

Comment:
There is indeed a 'pending' check. At the point where a PR is opened Travis reports how many statuses are to be applied. So if one has passed but the other hasn't, then we see a 'pending' and VB marks it as failed.

If there are multiple builds that run on a PR, then they'll all get their own status. So it won't work until all pass.